File: src/api/Libraries/Modules/ECDLink.DataAccessLayer/Entities/Navigation/Navigation.cs
using ECDLink.DataAccessLayer.Entities.Base;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace ECDLink.DataAccessLayer.Entities.Navigation
{
    [Table(nameof(Navigation))]
    public class Navigation : Navigation<Guid>
    {
    }

    public class Navigation<TKey> : EntityBase<TKey>
         where TKey : IEquatable<TKey>
    {
        public int Sequence { get; set; }
        public string Name { get; set; }
        public string Icon { get; set; }
        public string Route { get; set; }
        public string Description { get; set; }
    }
}

File: packages/core/src/schemas/navigation.ts
import * as Yup from 'yup';
import { NavigationDto } from '../models/dto/Navigation';

export const initialNavigationValues: NavigationDto = {
  sequence: 0,
  name: '',
  icon: '',
  route: '',
  description: '',
  permissions: [],
};

export const navigationSchema = Yup.object().shape({
  sequence: Yup.string(),
  name: Yup.string(),
  icon: Yup.string(),
  route: Yup.string(),
  description: Yup.string(),
});
